app/controllers/concerns/edge_cache_safety_check.rb

```diff
@@ -0,0 +1,17 @@
+module EdgeCacheSafetyCheck
+  extend ActiveSupport::Concern
+
+  CANNOT_USE_CURRENT_USER = "You may not use current_user in this cached code path.".freeze
+
+  def current_user
+    # In production, using current_user in a view which will be cached will result in a cache leak.
+    # https://docs.forem.com/technical-overview/architecture/#we-cache-many-content-pages-on-the-edge
+    return super unless @edge_caching_in_place
```

Comment:
This relies on an instance variable set by another concern. To me, this indicates that the two aren't really independent because one has knowledge about a very specific implementation detail of the other. Nothing good ever comes from this sort of implicit dependency.

I see several options:

1. We use the guaranteed thread-safe [request_store](https://github.com/steveklabnik/request_store) gem by Steve Klabnik:
    ```ruby
    #  app/controllers/concerns/caching_headers.rb
    RequestStore.store[:edge_caching_in_place] = true 

    # app/controllers/concerns/edge_cache_safety_check.rb
    return super unless RequestStore.store[:edge_caching_in_place]
2. You move this `current_user` implementation into the `CachingHeaders` concern but wrap it in a conditional (Ruby allows arbitrary code in class/module bodies):
    ```ruby
    unless Rails.env.production?
      def current_user
        ...
      end
    end
    ```

1. has the advantage of being future proof if you ever need to access the value of `RequestStore.store[:edge_caching_in_place]` anywhere else in the app. 2. is nice because everything caching related is stored in the same concern, without hidden dependencies. One could even consider combining both approaches once we actually do need to access this value in other parts of the code.

Reply:
I went with #1.